test/Npgsql.Tests/Support/PgServerMock.cs

```diff
@@ -77,6 +80,15 @@ internal async Task ReadMessageType(byte expectedCode)
             _readBuffer.Skip(len - 4);
         }
 
+        internal void BreakOnRead()
```

Comment:
There should probably just be a CloseSocket method exposed, plus the ReadBuffer itself - this way tests can do exactly what they want etc.